graphql-database-manager-dynamo/src/main/java/com/fleetpin/graphql/database/manager/dynamo/DynamoDb.java

```diff
@@ -890,6 +914,45 @@ public CompletableFuture<List<BackupItem>> takeBackup(String organisationId) {
 		});
 	}
 
+	@Override
+	public CompletableFuture<List<HistoryBackupItem>> takeHistoryBackup(String organisationId) {
+		if (this.classPath == null) {
+			throw new RuntimeException("classPath is required to obtain the tables' name to query the history table");
+		}
+
+		// Using reflections to loop through tables and get the tables' name
+		Set<Class<? extends Table>> tableObjects = new Reflections(classPath).getSubTypesOf(Table.class);
+		List<HistoryBackupItem> toReturn = Collections.synchronizedList(new ArrayList<HistoryBackupItem>());
+
+		Set<String> orgIdTypes = new HashSet<>(tableObjects.stream().map(obj -> organisationId + ":" + TableCoreUtil.table(obj)).collect(Collectors.toList()));
```

Comment:
toSet instead of toList to not need the new `HashSet`

Reply:
Refactored.